src/Core/Context/LegacyController.php

```diff
@@ -0,0 +1,281 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Context;
+
+use Media;
+use Module;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Traversable;
+
+class LegacyController
+{
+    // Dependency container
+    protected ContainerInterface $container;
+
+    public ?string $className;
+
+    /**
+     * List of CSS files.
+     *
+     * @var string[]
+     */
+    public array $css_files = [];
+
+    /**
+     * List of JavaScript files.
+     *
+     * @var string[]
+     */
+    public array $js_files = [];
+
+    // List of PHP errors.
+    public array $php_errors = [];
+
+    // If AJAX parameter is detected in request, set this flag to true.
+    public bool $ajax = false;
```

Comment:
I wonder if this one is useful 🤔

I believe in legacy controller this is what allows to detect if the process is an ajax request, but if we never initialize it, if the controller itself never changes the value to indicate to modules that they are in ajax and if we also never read it is it still useful? 🤔

Reply:
I do not think so. We can remove it